File: Merge-Strings-Alternately.java
class Solution 
{
    public String mergeAlternately(String word1, String word2) 
    {
        String result = "";
        int len1 = word1.length(), len2 = word2.length();

        for (int i = 0; i < len1 || i < len2; i++) 
        {
            if (i < len1) 
            {
                result += word1.charAt(i);
            }
            if (i < len2) 
            {
                result += word2.charAt(i);
            }
        }
        return result;
    }
}

File: Merge-Strings-Alternately.cpp
class Solution 
{
public:
    string mergeAlternately(string word1, string word2)
     {
        string result;
        int len1 = word1.size();
        int len2 = word2.size();

        for (int i = 0; i < len1 || i < len2; ++i) 
        {
            if (i < len1)
            {
                result += word1[i];
            }
            if (i < len2)
            {
                result += word2[i];
            }
        }
        return result;
     }
};